### ISSUE TITLE
Output of yolov8 for segmentation: how to save segmented image

### ISSUE THREAD

**Issue by Yasmina171** [ultralytics/ultralytics#17694]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I m using yolov8 for segmentation, i finetuned this model. and i want to use it in my application.
I deployed it with FastApi. 

My problem is that :  i want to use my model to detect objects (many objects) in an image after that i want to save the image after detction ( the segmented image ) in my database.  

how to do this in optimizing code because i m using cpu :: i want to know if  the results generated by the model have the segmented imag  or i have to  generat it ( because it takes a long time).

there is some of my code :: 

""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
    valid_image_data = [image.astype(np.uint8) for image in valid_image_data]

    loop = asyncio.get_running_loop()
    with ThreadPoolExecutor() as executor:
        results = await loop.run_in_executor(
            executor,
            lambda: model(valid_image_data, conf=conf_threshold, max_det=max_det, iou=iou, scale=scale)
        )

  objects_counts[image_id] = [len(result.boxes) for result in results]
 
"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""

(Sorry for my english).

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Yasmina171, thank you for reaching out to the Ultralytics community with your query! 🚀 

For new users, we recommend checking out the [Docs](https://docs.ultralytics.com) which provide a wealth of information on [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. Often, many common questions find their answers here.

If your question relates to output generation or optimizations using YOLOv8 segmentation, providing a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) would be immensely helpful. This assists us in better understanding your specific use case without guessing about your code or setup.

For deeper technical discussions and community support, join the Ultralytics family where it fits your style best! Engage in real-time chat on [Discord](https://ultralytics.com/discord) 🎧, explore thoughtful threads on [Discourse](https://community.ultralytics.com), and learn or share experiences on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

To ensure that the latest features or fixes are at your disposal, upgrade your `ultralytics` package. Confirm that all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) are met in a [**Python>=3.8**](https://www.python.org/) environment alongside [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO models can be executed across the following state-of-the-art environments, complete with essential dependencies like [CUDA](https://developer.nvidia.com/cuda), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/):

- **Notebooks** with free GPU: 
  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> 
  <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> 
  <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. Check out our [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. Find details in the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) 
  <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. Our CI tests validate the correct operation of YOLO's various [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across different platforms.

Thank you for your patience, and rest assured that an Ultralytics engineer will be here soon to provide further assistance! 😊
